backend/internal/server/server.go

```diff
@@ -4,43 +4,54 @@ import (
 	"fmt"
 	"log"
 	"net/http"
+	"time"
 
 	"github.com/gorilla/mux"
+	"github.com/ibeckermayer/teleport-interview/backend/internal/auth"
+	"github.com/ibeckermayer/teleport-interview/backend/internal/database"
 	"github.com/ibeckermayer/teleport-interview/backend/internal/handlers"
 )
 
 // Config is the top level config object.
 type Config struct {
-	port         int    // -port; default 8000
-	certFilePath string // -cert; default "../certs/localhost.crt"
-	keyFilePath  string // -key ; default "../certs/localhost.key"
+	port           int           // -port; default 8000
+	certFilePath   string        // -cert; default "../certs/localhost.crt"
+	keyFilePath    string        // -key ; default "../certs/localhost.key"
+	sessionTimeout time.Duration // -sesh; default 12h
+	env            string        // -env; default "prod"
 }
 
 // NewConfig creates a new server.Config
-func NewConfig(port int, certFilePath string, keyFilePath string) Config {
-	return Config{port, certFilePath, keyFilePath}
+func NewConfig(port int, certFilePath, keyFilePath string, sessionTimeout time.Duration, env string) Config {
+	return Config{port, certFilePath, keyFilePath, sessionTimeout, env}
 }
```

Comment:
nit: this constructor func seems redundant, if you export the fields of `Config`